### PR TITLE
EPMLSTRCMW-181: Wrong response message after attempt creating existing user

### PR DESCRIPTION
**Pull request by Dmitrii-Shevchenko**

### CODE REVIEW

File: services/src/main/scala/cromwell/pipeline/service/AuthService.scala [hunk 1]
```diff
@@ -37,24 +37,26 @@ class AuthService(userRepository: UserRepository, authUtils: AuthUtils)(implicit
...
-      lastName = request.lastName
-    )
-
-    userRepository.addUser(newUser).map { userId =>
-      val accessTokenContent = AccessTokenContent(userId)
-      val refreshTokenContent = RefreshTokenContent(userId, None)
-      getAuthResponse(accessTokenContent, refreshTokenContent, Instant.now.getEpochSecond)
+  def signUp(request: SignUpRequest): Future[Option[AuthResponse]] =
+    userRepository.getUserByEmail(request.email).flatMap {
+      case Some(_) => Future.failed(new RuntimeException(s"${request.email} already exists"))
```
Comment (OstapStrashevskii) [inline services/src/main/scala/cromwell/pipeline/service/AuthService.scala]: Why don't you make any custom Exception? There will be several Exceptions invoked by different authorization errors

File: services/src/test/scala/cromwell/pipeline/service/AuthServiceTest.scala [hunk 2]
```diff
@@ -56,19 +71,49 @@ class AuthServiceTest extends WordSpec with Matchers with MockFactory {
...
+        when(userRepository.getUserByEmail(user.email)).thenReturn(Future.successful(None))
+        when(userRepository.addUser(any[User])).thenReturn(Future.successful(user.userId))
+
+        authService.signUp(signUpRequest).map(response => response shouldBe Some(authResponse))
+      }
+
+      "return exception in case when registering with an existing email address" in {
+        when(userRepository.getUserByEmail(user.email)).thenReturn(Future.successful(Some(user)))
+
+        authService.signUp(signUpRequest).failed.map(e => e.getMessage shouldBe s"${user.email} already exists")
```
Comment (OstapStrashevskii) [inline services/src/test/scala/cromwell/pipeline/service/AuthServiceTest.scala]: you should replace it with whenready ;)

**Review by OstapStrashevskii — CHANGES_REQUESTED** [review]
Useful PR! Should you pay attention to my comments?

File: services/src/main/scala/cromwell/pipeline/service/AuthService.scala (hunk 1 above)
Comment (GrigoriiBerezin) [reply services/src/main/scala/cromwell/pipeline/service/AuthService.scala]: I agree with Ostap

**Comment by iliaonishchenko** [comment]
Conflicts.

File: services/src/test/scala/cromwell/pipeline/service/AuthServiceTest.scala (hunk 2 above)
Comment (Dmitrii-Shevchenko) [reply services/src/test/scala/cromwell/pipeline/service/AuthServiceTest.scala]: Done

File: services/src/main/scala/cromwell/pipeline/service/AuthService.scala (hunk 1 above)
Comment (Dmitrii-Shevchenko) [reply services/src/main/scala/cromwell/pipeline/service/AuthService.scala]: Done

**Review by GrigoriiBerezin — APPROVED** [review]
Nice

**Review by IaroslavTavchenkov — APPROVED** [review]
Seems clean and nice!
Thanks for you job!)

**Review by mnesvit — APPROVED** [review]

**Review by myazinn — APPROVED** [review]

**Pull request merged**